artemis/modules/subdomainenumeration.py

```diff
@@ -0,0 +1,121 @@
+            if response.ok:
+                soup = BeautifulSoup(response.text, 'html.parser')
+                th_tags = soup.find_all('th')
+                for th_tag in th_tags:
+                    td_tag = th_tag.find_next_sibling('td')
+                    if td_tag is not None:
+                        url = td_tag.get_text()
+                        subdomains.add(url)
+        except requests.exceptions.RequestException:
+            self.log.exception("Unable to obtain information from rapiddns.io for domain %s", domain)
+        return subdomains
+
+    def get_subdomains_from_jldc(self, current_task: Task) -> Set[str]:
+        domain = current_task.get_payload("domain")
+        subdomains: Set[str] = set()
+        try:
+            response = requests.get(f"https://jldc.me/anubis/subdomains/{domain}")
+            if response.ok:
+                urls = eval(response.text)
+                subdomains.update(urls)
+        except requests.exceptions.RequestException:
+            self.log.exception("Unable to obtain information from jldc.me for domain %s", domain)
+        return subdomains
+
+    def get_subdomains_from_subfinder(self, current_task: Task) -> Set[str]:
+        domain = current_task.get_payload("domain")
+        subdomains: Set[str] = set()
+        try:
+            result = check_output_log_on_error(
+                [
+                    "subfinder", 
+                    "-d", 
+                    domain, 
+                    "-silent",
+                    "-all"
+                    ],
+                    self.log
+            )
+            subdomains.update(result.decode().splitlines())
+        except Exception:
+            self.log.exception("Unable to obtain information from subfinder for domain %s", domain)
+        return subdomains
+
+    def get_subdomains_from_amass(self, current_task: Task) -> Set[str]:
+        domain = current_task.get_payload("domain")
+        subdomains: Set[str] = set()
+        try:
+            result = check_output_log_on_error(
+                [
+                    "amass",
+                    "enum",
+                    "-d",
+                    domain,
+                    "-silent"
+                ],
+                self.log
+            )
+            subdomains.update(result.decode().splitlines())
+        except Exception:
+            self.log.exception("Unable to obtain information from amass for domain %s", domain)
+        return subdomains
+
+
+    def run(self, current_task: Task) -> None:
+        domain = current_task.get_payload("domain")
+        subdomains = (
+            self.get_subdomains_from_subfinder(current_task) |
+            self.get_subdomains_from_amass(current_task) |
+            self.get_subdomains_from_rapid(current_task) |
+            self.get_subdomains_from_jldc(current_task)
+        )
+
+        for subdomain in subdomains:
+            if is_subdomain(subdomain, domain):
+                task = Task(
```

Comment:
this way subdomain enumeration will run on all subdomains returned from this module

Reply:
hi it's intentional because sometimes the testing subdomains have another layer of subdomains.

for eg. 
test1.testing.example.com
test2.testing.example.com

Reply:
IMO it's better to do this in one pass if possible as this may be a lot faster